### DISCUSSION TITLE
Should constants be upper case?

### DISCUSSION THREAD

**Discussion by David-Worboys** [vlang/v#15791] [Ideas]
I am used to constants being uppercase but V likes lowercase constants as below

```
const envelope_tags = 1
const application_tags = 2
```
And I find this can be confusing to me as code below illustrates

```
if iptc.envelope_tags.len == 0 {
		iptc.iptc_tag_map[envelope_tags] = json_utils.read_tags('iptcenvelopetags.json')
	}

```
Looking over a block of code the envelope_tags constant does not leap out to me but blends in to the rest of the code.

My view is that constants should be upper case, thoughts appreciated!

**Comment by YamilBracho** [comment]
Remember  this is a convention (inherited from C language), so you can name  your constants for example, starts with k_ or something like that and trhat way
it would be easier to recognize constants in your code and if your editor offers sone intellisense technology you will see all symbols starts with k_
in a group...

HTH,
Yamil

**Comment by David-Worboys** [comment]
Agree Yaml, one could prefix constants - and I have done so in other languages that lack them -and my C background leads me to expect constants to be uppercase, so yes that convention is strong with me.  

I do feel that constants are special and need to be displayed in a way different from variables to make it clear that one is using a constant. To me, uppercase seems to be the simplest way to do that.

**Reply by JalonSolov** [reply]
V uses uppercase for something else... struct and user-created type names.  If you also used uppercase for consts, it would introduce more confusion, not reduce it.

It could be worse... in Go, for example, uppercase means a public symbol, while lowercase means private.  :-\
